code/modules/ninja/ninja_explosive.dm

```diff
@@ -10,38 +10,64 @@
 	name = "spider charge"
 	desc = "A modified C-4 charge supplied to you by the Spider Clan.  Its explosive power has been juiced up, but only works in one specific area."
 	boom_sizes = list(4, 8, 12)
+	///The mob that has planted the charge
 	var/mob/detonator = null
+	///The only area that the charge is allowed to be planted, and detonated in
+	var/detonation_area = null
 
-/obj/item/grenade/c4/ninja/afterattack(atom/movable/AM, mob/user, flag)
-	var/datum/antagonist/ninja/ninja_antag = user.mind.has_antag_datum(/datum/antagonist/ninja)
+/obj/item/grenade/c4/ninja/Initialize(mapload)
+	. = ..()
+	var/mob/living/user = loc
```

Comment:
This seems very fragile, I was hoping that you'd be able to pass in the ninja datum as an arg - or in the post equip of the outfit or something idk